### PR TITLE
Add rating class and commands

### LINKED ISSUE #97: Add Rating class

### PR DESCRIPTION
**Pull request by CelestineTan03**
Closes #97

### CODE REVIEW

File: src/main/java/seedu/staffsnap/model/interview/Interview.java [hunk 1]
```diff
@@ -48,19 +60,30 @@ public boolean equals(Object other) {
...
-        return type.equals(otherInterview.type);
+        return type.equals(otherInterview.type)
+                && rating.equals(otherInterview.rating);
+    }
+
+    /**
+     * Returns true if the applicant has an interview with the same interview type.
+     */
+    public boolean isSameInterview(List<Interview> otherInterviews) {
+        return otherInterviews.stream().anyMatch(interview -> interview.getType().equals(getType()));
```
Comment (craigtonlian) [inline src/main/java/seedu/staffsnap/model/interview/Interview.java:71]: `isSameInterview` seems to imply we're comparing an instance of Interview with another instance of Interview. 

Perhaps we can rename it has `isContainedIn` to better reflect the method signature? Any better name suggestions? 😆

File: src/main/java/seedu/staffsnap/model/interview/Interview.java [hunk 2]
```diff
@@ -14,15 +19,18 @@ public class Interview implements Comparable<Interview> {
...
      * @param type A valid interview type.
      */
-    public Interview(String type) {
-        requireNonNull(type);
+    public Interview(String type, Rating rating) {
+        requireAllNonNull(type);
         checkArgument(isValidType(type), MESSAGE_CONSTRAINTS);
         this.type = type;
+        this.rating = rating;
     }
```
Comment (craigtonlian) [inline src/main/java/seedu/staffsnap/model/interview/Interview.java:34]: Why doesn't `rating` require a null check?

**Review by craigtonlian — APPROVED** [review]
Very thorough work! Appreciate the meticulous eye for detail 👁️ 👁️ 

Perhaps can update the UG correspondingly? Thanks, Celestine!

File: src/main/java/seedu/staffsnap/model/interview/Interview.java (hunk 1 above)
Comment (CelestineTan03) [reply src/main/java/seedu/staffsnap/model/interview/Interview.java:71]: That's a very good suggestion 😄

File: src/main/java/seedu/staffsnap/model/interview/Rating.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Represents an Interview's rating in the applicant book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRating(String)}
+ */
+public class Rating {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Rating should be a number between 0.0 and 10.0 to 1 decimal place";
+    public static final String VALIDATION_REGEX = "^(10|10.0|\\d{1}(\\.\\d{1})?)$";
```
Comment (craigtonlian) [inline src/main/java/seedu/staffsnap/model/interview/Rating.java:14]: Really cool regex ❤️

**Review by craigtonlian — COMMENTED** [review]
Lgtm!

**Pull request merged**